src/server/gc_worker/rawkv_compaction_filter.rs

```diff
@@ -0,0 +1,455 @@
+}
+
+impl Drop for RawCompactionFilter {
+    // NOTE: it's required that `CompactionFilter` is dropped before the compaction result
+    // becomes installed into the DB instance.
+    fn drop(&mut self) {
+        self.raw_gc_mvcc_deletions();
+
+        self.engine.sync_wal().unwrap();
+
+        self.switch_key_metrics();
+        self.flush_metrics();
+    }
+}
+
+impl CompactionFilter for RawCompactionFilter {
+    fn featured_filter(
+        &mut self,
+        level: usize,
+        key: &[u8],
+        sequence: u64,
+        value: &[u8],
+        value_type: CompactionFilterValueType,
+    ) -> CompactionFilterDecision {
+        match self.do_filter(level, key, sequence, value, value_type) {
+            Ok(decision) => decision,
+            Err(e) => {
+                warn!("compaction filter meet error: {}", e);
+                GC_COMPACTION_FAILURE.with_label_values(&["filter"]).inc();
+                CompactionFilterDecision::Keep
+            }
+        }
+    }
+}
+
+impl RawCompactionFilter {
+    fn new(
+        engine: RocksEngine,
+        safe_point: u64,
+        gc_scheduler: Scheduler<GcTask<RocksEngine>>,
+        ts: u64,
+        _context: &CompactionFilterContext,
+        regions_provider: (u64, Arc<dyn RegionInfoProvider>),
+    ) -> Self {
+        // Safe point must have been initialized.
+        assert!(safe_point > 0);
+        debug!("gc in compaction filter"; "safe_point" => safe_point);
+        RawCompactionFilter {
+            safe_point,
+            engine,
+            gc_scheduler,
+            current_ts: ts,
+            mvcc_key_prefix: vec![],
+            mvcc_deletions: Vec::with_capacity(DEFAULT_DELETE_BATCH_COUNT),
+            regions_provider,
+
+            versions: 0,
+            filtered: 0,
+            total_versions: 0,
+            total_filtered: 0,
+            mvcc_rollback_and_locks: 0,
+            orphan_versions: 0,
+            versions_hist: MVCC_VERSIONS_HISTOGRAM.local(),
+            filtered_hist: GC_DELETE_VERSIONS_HISTOGRAM.local(),
+        }
+    }
+
+    fn do_filter(
+        &mut self,
+        _start_level: usize,
+        key: &[u8],
+        _sequence: u64,
+        value: &[u8],
+        value_type: CompactionFilterValueType,
+    ) -> Result<CompactionFilterDecision, String> {
+        if !key.starts_with(keys::DATA_PREFIX_KEY) {
+            return Ok(CompactionFilterDecision::Keep);
+        }
+
+        // remove prefix 'z'
+        let current_key = keys::origin_key(key);
+        let key_mode = ApiV2::parse_key_mode(current_key);
+
+        // not RawKV or targetValue
+        if key_mode != KeyMode::Raw || value_type != CompactionFilterValueType::Value {
+            return Ok(CompactionFilterDecision::Keep);
+        }
+
+        let (mvcc_key_prefix_vec, commit_ts_opt) =
+            ApiV2::decode_raw_key(&Key::from_encoded_slice(current_key), true).unwrap();
+        let mvcc_key_prefix = mvcc_key_prefix_vec.as_slice();
```

Comment:
It doesn't seems necessary to keep this slice in a variable here. It seems ok to just use `mvcc_key_prefix_vec`, and the `_vec` suffix in the name can be removed.